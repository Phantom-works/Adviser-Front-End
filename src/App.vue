<template>
  <div>
    <GameAdviserHeader text="Game Adviser" />
    <CategoryList :categories="categories" @DeleteCategory="DeleteCategory" />
  </div>

  <div>
    <RoundedButton text="hello world" />
  </div>
</template>

<script lang="ts" setup>
import RoundedButton from './components/default/DefaultButton.vue'
import GameAdviserHeader from '@/components/GameAdviserHeader.vue'
import CategoryList from '@/components/category/GameCategoryList.vue'
import type Category from '@/BLL/Category'

import { ref } from 'vue'
const DeleteCategory = (_id: number) => {
  categories.value = categories.value.filter((category) => category.id !== _id)
}

let categories = ref<Category[]>([
  {
    id: 1,
    title: 'sandbox'
  },
  {
    id: 2,
    title: 'roguelike'
  },
  {
    id: 3,
    title: 'puzzle'
  },
  {
    id: 4,
    title: 'open world'
  }
])
</script>

<style scoped></style>
