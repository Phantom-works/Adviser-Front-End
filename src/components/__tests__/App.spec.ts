import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import App from '../../App.vue'

describe('App Snapshots', () => {
  // Arrange

  it('Snapshot App Default', () => {
    //Arrange
    // Act
    const wrapper = mount(App, {
      props: {}
    })
    // Assert
    expect(wrapper.html()).toMatchSnapshot()
  })
})
