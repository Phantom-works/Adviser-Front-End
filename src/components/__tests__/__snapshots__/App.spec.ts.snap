// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Snapshots > Snapshot App Default 1`] = `
"<div data-v-7a7a37b1=\\"\\" class=\\"game-container\\">
  <div data-v-7a7a37b1=\\"\\">
    <div data-v-8d32426d=\\"\\" data-v-7a7a37b1=\\"\\" class=\\"categories-container\\">
      <div data-v-15dafa4e=\\"\\" data-v-8d32426d=\\"\\" class=\\"input-container\\"><input data-v-15dafa4e=\\"\\" class=\\"rounded-input\\" placeholder=\\"Enter text here\\" type=\\"text\\"><button data-v-dec1f997=\\"\\" data-v-15dafa4e=\\"\\" class=\\"button\\">+</button></div>
    </div>
  </div>
  <div data-v-7a7a37b1=\\"\\">
    <div data-v-cc4cce50=\\"\\" data-v-7a7a37b1=\\"\\" class=\\"game-list\\"></div>
  </div>
</div>"
`;
